pam/integration-tests/gdm-integration_test.go

```diff
@@ -0,0 +1,389 @@
+package main_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/msteinert/pam/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/testutils"
+	grouptests "github.com/ubuntu/authd/internal/users/localgroups/tests"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/gdm_test"
+	"github.com/ubuntu/authd/pam/internal/pam_test"
+	"github.com/ubuntu/authd/pam/internal/proto"
+)
+
+func init() {
+	gdm.AdvertisePamExtensions([]string{gdm.PamExtensionCustomJSON})
+}
+
+const (
+	exampleBrokerName = "ExampleBroker"
+	localBrokerName   = "local"
+	ignoredBrokerName = "<ignored-broker>"
+
+	passwordAuthID = "password"
+	fido1AuthID    = "fidodevice1"
+	phoneAck1ID    = "phoneack1"
+)
+
+func TestGdmModule(t *testing.T) {
+	t.Parallel()
+
+	// This happens when GO_WANT_HELPER_PROCESS is defined, in such case let's just skip.
+	if daemonPath == "" {
+		t.Skip("Test cannot run without an example daemon")
+	}
```

Comment:
Same thing here. These tests will not be called when we set `GO_WANT_HELPER_PROCESS`. If they were, it means we messed up the mocking and we should take another look at the set up.